DICOMPlugins/DICOMTID1500Plugin.py

```diff
@@ -157,6 +292,899 @@ def getDateTime(self, uid):
+    col = tableNode.AddColumn()
+    col.SetName("FindingType")
+    col = tableNode.AddColumn()
+    col.SetName("FindingSite")
+    col = tableNode.AddColumn()
+    col.SetName("PolyLine")
+
+    for i,p in enumerate(point_infos):
+      # get values 
+      tracking_identifier = p['TrackingIdentifier']
+      tracking_uid = p['TrackingUID']
+      finding_type = p['FindingType']
+      finding_site = p['FindingSite'][0] # check this later. 
+      polyline = p["polyline"]
+      polyline = [str(np.round(f,2)) for f in polyline]
+      polyline_str = f"({', '.join(polyline)})"
+      # add tracking info and finding site info 
+      rowIndex = tableNode.AddEmptyRow()
+      tableNode.SetCellText(rowIndex, 0, tracking_identifier)
+      tableNode.SetCellText(rowIndex, 1, finding_type[2]) # CodeMeaning
+      tableNode.SetCellText(rowIndex, 2, finding_site[2]) # CodeMeaning
+      # add polyline
+      tableNode.SetCellText(rowIndex, 3, polyline_str) 
+
+    return tableNode 
+
+  def extractBboxMetadataToVtkTableNode(self, srFileName): 
+    """
+    Extracts the bounding box metadata from the SR using highdicom, 
+    and creates a table node. 
+    """
+    # read SR using highdicom 
+    sr = hd.sr.srread(srFileName)
+
+    # get the referenced SeriesInstanceUID 
+    referenced_series_instance_uid = sr.CurrentRequestedProcedureEvidenceSequence[0].ReferencedSeriesSequence[0].SeriesInstanceUID
+
+    # will store the info needed for table too. 
+    poly_infos = [] 
+
+    # First get the planar roi measurement gorups 
+    groups = sr.content.get_planar_roi_measurement_groups()
+
+    for group in groups: 
+
+      # Get the tracking ids 
+      tracking_identifier = group.tracking_identifier
+      tracking_uid = group.tracking_uid
+
+      # Get the findings and finding_sites 
+      finding_type = [group.finding_type.CodeValue, group.finding_type.CodingSchemeDesignator, group.finding_type.CodeMeaning]
+      finding_sites = []
+      for finding_site in group.finding_sites:
+        if hasattr(finding_site, 'ConceptCodeSequence') and finding_site.ConceptCodeSequence:
+          finding_sites.append([finding_site.ConceptCodeSequence[0].CodeValue, 
+                                finding_site.ConceptCodeSequence[0].CodingSchemeDesignator,
+                                finding_site.ConceptCodeSequence[0].CodeMeaning])
+
+      # if (group.reference_type.meaning == "Image Region"): # why meaning instead of CodeMeaning? 
+      if (group.reference_type.value=="111030" and \
+          group.reference_type.scheme_designator=="DCM" and \
+          group.reference_type.meaning=="Image Region"):
+
+        roi = group.roi # should exist if Image Region is present      
+        try: 
+          referenced_sop_instance_uid = roi.ContentSequence[0].referenced_sop_instance_uid
+        except: 
+          print('Cannot access referenced SOPInstanceUID')
+
+        if (roi.GraphicType=="POLYLINE"):
+          bbox = roi.GraphicData 
```

Comment:
Might be slightly easier to use `roi.value`, which will correctly format the array for you as a (n x 2) or (n x 3) numpy array, for SCOORD or SCOORD3D respectively

Reply:
Using` roi.value` now, fixed in [branch](https://github.com/deepakri201/QuantitativeReporting/tree/add_planar_annot).